shared-libs/transitions/src/transitions/create_user_for_contacts.js

```diff
@@ -154,30 +154,49 @@ module.exports = {
       !contactType
       || !contactTypeUtils.isPersonType(contactType)
       || !doc.user_for_contact
-      || !doc.user_for_contact.replace
     ) {
       return false;
     }
 
-    return !!Object
-      .values(doc.user_for_contact.replace)
-      .find(({ status }) => status === USER_CREATION_STATUS.READY);
+    if (doc.user_for_contact.replace) {
+      return !!Object
+        .values(doc.user_for_contact.replace)
+        .find(({ status }) => status === USER_CREATION_STATUS.READY);
+    }
+
+    if (doc.user_for_contact.add) {
+      return doc.user_for_contact.add.status === USER_CREATION_STATUS.READY;
+    }
+
+    return false;
   },
   onMatch: async change => {
-    const usersToReplace = getUsersToReplace(change.doc);
-    const replaceUsers = usersToReplace.map(user => replaceUser(change.doc, user));
-    const replacementResults = await Promise.allSettled(replaceUsers);
-    const errors = replacementResults
-      .filter(({ status }) => status === 'rejected')
-      .map(({ reason }) => reason);
-    if (errors.length) {
-      throw {
-        changed: true,
-        message: errors
-          .map(error => error.message || JSON.stringify(error))
-          .join(', '),
-      };
+    const isReplacingUser = Boolean(change.doc.user_for_contact.replace);
+    if (isReplacingUser) {
+      const usersToReplace = getUsersToReplace(change.doc);
+      const replaceUsers = usersToReplace.map(user => replaceUser(change.doc, user));
+      const replacementResults = await Promise.allSettled(replaceUsers);
```

Comment:
I think we should work the add-user processing into this same `Promise.allSettled` (along with the error handling) so that a failure in the user-replace does not prevent the user-add from running.

Reply:
The `Promise.allSettled` makes sense for the replacement because we got an array of promises creating potentially many users but in this case a simple `try`/`catch` should be enough, no need to go through the gymnastics of creating an array of promises and looping through an array of errors when we know we'll always have just one element.

I agree with you we should be consistent with error handling though and I made a few changes to keep the behavior consistent with the replace code and extracted the couple of lines there into a `addUser` function

Reply:
I un-resolved this conversation since I think it depends on the outcome of the earlier thread regarding support for  adding and replacing users for the same contact.  If we decide we do not want to support that, we can re-resolve this conversation.  If we end up wanting to support adding and replacing at the same time, we should revisit this logic....